evutil.c

```diff
@@ -480,13 +480,37 @@ evutil_make_listen_socket_reuseable(evutil_socket_t sock)
 int
 evutil_make_listen_socket_reuseable_port(evutil_socket_t sock)
 {
-#if defined __linux__ && defined(SO_REUSEPORT)
-	int one = 1;
-	/* REUSEPORT on Linux 3.9+ means, "Multiple servers (processes or
-	 * threads) can bind to the same port if they each set the option. */
-	return setsockopt(sock, SOL_SOCKET, SO_REUSEPORT, (void*) &one,
-	    (ev_socklen_t)sizeof(one));
+#if defined(__FreeBSD__) && __FreeBSD__ >= 12 && defined(SO_REUSEPORT_LB)
+	/* FreeBSD 12 introduced a new socket option named SO_REUSEPORT_LB
+	 * with the capability of load balancing, it's the equivalent of
+	 * the SO_REUSEPORTs on Linux and DragonFlyBSD. */
+	int enabled = 1;
+	return setsockopt(sock, SOL_SOCKET, SO_REUSEPORT_LB, (void*)&enabled,
+	    (ev_socklen_t)sizeof(enabled));
+#elif (defined __linux__ || defined __DragonFly__ || defined __sun) && defined(SO_REUSEPORT)
```

Comment:
What a strange syntax, but it works. Let's use proper `defined()` everywhere.

Reply:
Done.